pkg/resourcequotas/resourcequotas.go

```diff
@@ -0,0 +1,198 @@
+package resourcequotas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1Typed "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// Builder provides struct for resource quotas containing connection to the cluster.
+type Builder struct {
+	// Resource Quota definition
+	Definition *corev1.ResourceQuota
+
+	// Created resource quota object
+	Object *corev1.ResourceQuota
+
+	// Used in functions that define or mutate deployment definition. errorMsg is processed before the deployment
+	// object is created.
+	errorMsg  string
+	apiClient corev1Typed.CoreV1Interface
+}
+
+// NewBuilder creates a new resource quota builder.
+func NewBuilder(apiClient corev1Typed.CoreV1Interface, name, nsname string,
```

Comment:
I think for consistency it makes more sense to take `*clients.Settings` and then take the `CoreV1Interface` from it for the builder

Reply:
Done.